From a40ce1b0e7851b881e8751e22963a93335ec03b6 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 17 Mar 2020 19:38:39 +0800
Subject: [PATCH] kernel/params.c: Call param_sysfs_init() early when
 CONFIG_ROCKCHIP_THUNDER_BOOT=y

usb_init() depends on param_sysfs_init() to init module_kset.
But usb_init() and param_sysfs_init() are both in the same
subsys_initcall level.

Fix this by move param_sysfs_init() to the arch_initcall_sync level.

Change-Id: Iccc44152c716959d98ed97ea379e0fc4378e185d
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 kernel/params.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/params.c b/kernel/params.c
index ce89f757e6da..d538bc469311 100644
--- a/kernel/params.c
+++ b/kernel/params.c
@@ -955,6 +955,10 @@ static int __init param_sysfs_init(void)
 
 	return 0;
 }
+#ifdef CONFIG_ROCKCHIP_THUNDER_BOOT
+arch_initcall_sync(param_sysfs_init);
+#else
 subsys_initcall(param_sysfs_init);
+#endif
 
 #endif /* CONFIG_SYSFS */
-- 
2.35.3

